File: benchmarks/go/string_scan.go
package main

import "os"

func main() {
	text := []byte("TezzNative makes native tools readable and fast.")
	acc := 0
	for i := 0; i < 60000; i++ {
		for j, ch := range text {
			acc = (acc + ((int(ch) * (j + 1)) & 65535)) & 1048575
		}
	}
	if acc == -1 {
		os.Exit(1)
	}
}

File: benchmarks/python/string_scan.py
text = b"TezzNative makes native tools readable and fast."
reps = 60_000
acc = 0
for _ in range(reps):
    for j, ch in enumerate(text):
        acc = (acc + ((ch * (j + 1)) & 65_535)) & 1_048_575
if acc == -1:
    raise SystemExit(1)
